test/jobs/dspace_publication_results_job_test.rb

```diff
@@ -2,20 +2,35 @@
 
 class DspacePublicationResultsJobTest < ActiveJob::TestCase
   def setup
-    @good_thesis = theses(:one)
+    @good_thesis = theses(:bachelor)
     @bad_thesis = theses(:two)
+    @bad_checksum = theses(:doctor)
     @no_handle_thesis = theses(:coauthor)
     @invalid_status_thesis = theses(:with_note)
+    @valid_with_no_local_files = theses(:one)
     Aws.config[:sqs] = {
       stub_responses: {
         receive_message: [
           {
             messages: [
               # success
-              { message_id: 'id1', receipt_handle: 'handle1', body: '{"ResultType": "success", "ItemHandle": "http://example.com/handle/123123123", "lastModified": "Thu Sep 09 17: 56: 39 UTC 2021"}',
+              { message_id: 'id1', receipt_handle: 'handle1',
+                body: '{"ResultType": "success", "ItemHandle": "http://example.com/handle/123123123", "lastModified": "Thu Sep 09 17: 56: 39 UTC 2021", "Bitstreams": [{ "BitstreamName": "a_pdf.pdf", "BitstreamUUID": "fakeuuidshhhhh",  "BitstreamChecksum": { "value": "2800ec8c99c60f5b15520beac9939a46", "checkSumAlgorithm": "MD5"}}]}',
                 message_attributes: { 'PackageID' => { string_value: "etd_#{@good_thesis.id}", data_type: 'String' },
                                       'SubmissionSource' => { string_value: 'ETD', data_type: 'String' } } },
 
+              # success but invalid checksum
+              { message_id: 'id1a', receipt_handle: 'handle1a',
+                body: '{"ResultType": "success", "ItemHandle": "http://example.com/handle/123123123", "lastModified": "Thu Sep 09 17: 56: 39 UTC 2021", "Bitstreams": [{ "BitstreamName": "a_pdf.pdf", "BitstreamUUID": "fakeuuidshhhhh",  "BitstreamChecksum": { "value": "borkedchecksum", "checkSumAlgorithm": "MD5"}}]}',
+                message_attributes: { 'PackageID' => { string_value: "etd_#{@bad_checksum.id}", data_type: 'String' },
+                                      'SubmissionSource' => { string_value: 'ETD', data_type: 'String' } } },
+
+              # success but thesis no longer has files locally
+              { message_id: 'id1a', receipt_handle: 'handle1a',
+                body: '{"ResultType": "success", "ItemHandle": "http://example.com/handle/123123123", "lastModified": "Thu Sep 09 17: 56: 39 UTC 2021", "Bitstreams": [{ "BitstreamName": "a_pdf.pdf", "BitstreamUUID": "fakeuuidshhhhh",  "BitstreamChecksum": { "value": "borkedchecksum", "checkSumAlgorithm": "MD5"}}]}',
```

Comment:
Even though it doesn't affect the test result, I think this message should have a 'real' checksum value to avoid confusion.

Reply:
Good point. I agree.